src/Input.jsx

```diff
@@ -56,18 +58,23 @@ const Input = React.createClass({
 
   render() {
     const placeholder = this.props.placeholder || '';
-    let size = this.props.value.length === 0 ?
-      placeholder.length :
-      this.props.value.length;
+    let extraProps = {};
+
+    if (this.props.inputAutoSize){
+        extraProps["size"] = (this.props.value.length === 0 ?
```

Comment:
You are right, with null is better, I forgot that react is so smart. I would like to add other props, for example "maxlength" is a must, for me at least it ruins CSS if you type or paste a large text, when using the "size" attribute.